### PR TITLE
Update dependencies

### LINKED ISSUE #230: Push a new release for google-api-client 1.28.0
@wsh Would you release new version for Java (`com.google.cloud.datastore:datastore-v1-proto-client:1.6.0`)? I see @elharo committed change to updated google-api-client 1.28.0 but it has not been released yet.

(background: I was checking https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/699 )

### LINKED ISSUE #232: google-api-client 1.29.2
google-api-client 1.30.2 is the latest release of google-api-client. 1.29.2 is the latest without changing artifact ID of google-http-client-jackson

image

I tried to upgrade the version to 1.30.2 but encountered 2 missing dependencies:
```
[ERROR] Failed to execute goal on project datastore-v1-proto-client: Could not resolve dependencies for project com.google.cloud.datastore:datastore-v1-proto-client:jar:1.6.1: The following artifacts could not be resolved:
 com.google.http-client:google-http-client-jackson:jar:1.30.2,
 com.google.oauth-client:google-oauth-client:jar:1.30.2:
Could not find artifact com.google.http-client:google-http-client-jackson:jar:1.30.2 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```

### PR DESCRIPTION
**Pull request by elharo**
@clement @suztomo @chingor13 

fixes #232 
fixes #230

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -69,43 +69,37 @@
...
         <artifactId>google-oauth-client</artifactId>
-        <version>1.28.0</version>
+        <version>1.30.4</version>
       </dependency>
 
       <dependency>
         <groupId>com.google.api-client</groupId>
         <artifactId>google-api-client</artifactId>
-        <version>1.28.0</version>
+        <version>1.30.5</version>
```
Comment (suztomo) [inline java/pom.xml]: Make these numbers (such as 1.30.5) variable(s)

File: java/pom.xml [hunk 2]
```diff
@@ -69,43 +69,37 @@
...
         <groupId>com.google.http-client</groupId>
         <artifactId>google-http-client</artifactId>
-        <version>1.28.0</version>
-        <exclusions>
-          <exclusion>
-            <artifactId>guava-jdk5</artifactId>
-            <groupId>com.google.guava</groupId>
-          </exclusion>
-        </exclusions>
+        <version>1.32.1</version>
```
Comment (netdpb) [inline java/pom.xml]: Should this use a BOM instead?

File: java/pom.xml (hunk 1 above)
Comment (elharo) [reply java/pom.xml]: The only ones that are duped are now pulled in via a BOM.

File: java/pom.xml (hunk 2 above)
Comment (elharo) [reply java/pom.xml]: done

**Pull request merged**